lib/codecs/src/encoding/format/csv.rs

```diff
@@ -0,0 +1,143 @@
+use crate::encoding::BuildError;
+use bytes::{BufMut, BytesMut};
+use lookup::lookup_v2::OwnedValuePath;
+use tokio_util::codec::Encoder;
+use vector_core::{
+    config::DataType,
+    event::{Event, Value},
+    schema,
+};
+
+/// Config used to build a `CsvSerializer`.
+#[crate::configurable_component]
+#[derive(Debug, Clone, Default)]
+pub struct CsvSerializerConfig {
+    /// The CSV fields.
+    pub fields: Vec<OwnedValuePath>,
```

Comment:
I'm not wholly convinced by this configuration option, we have `encoding.only_fields` configuration option for Sinks today which covers much of the same purpose.

Reply:
Yes, I also considered this config. But in order to use `only_config` from the transformer, we need to build the serializer with transformer as an argument. That seems a little confusing, and would also introduce a circular import.

Reply:
We could make this config deeper in `csv` context, like `encoding.csv.fields`, maybe that seems better?